common/flatpak-context.c

```diff
@@ -582,6 +583,18 @@ flatpak_context_set_persistent (FlatpakContext *context,
   return TRUE;
 }
 
+static gboolean
+flatpak_context_set_data_dir (FlatpakContext *context,
+                              const char     *path,
+                              GError        **error)
+{
+  if (!flatpak_validate_path_characters (path, error))
+    return FALSE;
+
+  context->data_dir = g_strdup(path);
```

Comment:
I don't see were `data_dir` is freed when the context is freeed in `
flatpak_context_free()`

Reply:
Nor where it is freeed if the value is being overridden.